kpet/data.py

```diff
@@ -391,16 +391,33 @@ def __init__(self, data):
         """
         Initialize a host type.
         """
+        # TODO Drop the old schema once kpet-db is switched to the new one
+        def inherit(data):
+            if "tasks" in data:
+                data["preboot_tasks"] = data.pop("tasks")
+            return data
         super().__init__(
             "host type",
-            Struct(optional=dict(
-                ignore_panic=Boolean(),
-                hostRequires=String(),
-                hostname=String(),
-                partitions=String(),
-                kickstart=String(),
-                tasks=String(),
-            )),
+            Succession(
+                Struct(optional=dict(
+                    ignore_panic=Boolean(),
+                    hostRequires=String(),
+                    hostname=String(),
+                    partitions=String(),
+                    kickstart=String(),
+                    tasks=String(),
+                )),
+                inherit,
+                Struct(optional=dict(
+                    ignore_panic=Boolean(),
+                    hostRequires=String(),
+                    hostname=String(),
+                    partitions=String(),
+                    kickstart=String(),
+                    preboot_tasks=String(),
+                    postboot_tasks=String(),
+                )),
```

Comment:
Maybe I don't understand what you're trying to do, but shouldn't the constructor also accept `{pre,post}boot_tasks`? Why the first parameter to `Succession`?

Reply:
The first parameter to succession is the original schema, the second is a conversion function, and the third is the resulting schema. I.e. we convert from the first schema to the second schema.

Reply:
But, does that mean that the incoming data has to pass the first schema, and then pass the second schema after the conversion in the middle?

If so, doesn't that mean that the incoming schema _cannot_ have `preboot_tasks` or `postboot_tasks`? And if so, wouldn't we want

I don't have the whole flow in my head so I'm not sure my assumptions are correct, so feel free to ignore.

Reply:
The `Succession` would accept any of the schemas, but would convert any of them to the last one, through the intermediate ones. I.e. kpet would accept host types with either "tasks", or "preboot_tasks" and "postboot_tasks".